src/servers/src/http/admin.rs

```diff
@@ -0,0 +1,227 @@
+            StatusCode::INTERNAL_SERVER_ERROR,
+            "Internal Server Error".to_string(),
+        )
+            .into_response(),
+    }
+}
+
+pub async fn maintenance_status_handler(State(state): State<AdminState>) -> impl IntoResponse {
+    match state
+        .maintenance_handler
+        .handle("/maintenance/status", Method::GET, &HashMap::new())
+        .await
+    {
+        Ok((status, body)) => (status, body).into_response(),
+        Err(_) => (
+            StatusCode::INTERNAL_SERVER_ERROR,
+            "Internal Server Error".to_string(),
+        )
+            .into_response(),
+    }
+}
+
+pub async fn maintenance_enable_handler(State(state): State<AdminState>) -> impl IntoResponse {
+    match state
+        .maintenance_handler
+        .handle("/maintenance/enable", Method::POST, &HashMap::new())
+        .await
+    {
+        Ok((status, body)) => (status, body).into_response(),
+        Err(_) => (
+            StatusCode::INTERNAL_SERVER_ERROR,
+            "Internal Server Error".to_string(),
+        )
+            .into_response(),
+    }
+}
+
+pub async fn maintenance_disable_handler(State(state): State<AdminState>) -> impl IntoResponse {
+    match state
+        .maintenance_handler
+        .handle("/maintenance/disable", Method::POST, &HashMap::new())
+        .await
+    {
+        Ok((status, body)) => (status, body).into_response(),
+        Err(_) => (
+            StatusCode::INTERNAL_SERVER_ERROR,
+            "Internal Server Error".to_string(),
+        )
+            .into_response(),
+    }
+}
+
+pub async fn procedure_status_handler(State(state): State<AdminState>) -> impl IntoResponse {
+    match state
+        .procedure_handler
+        .handle("/procedure-manager/status", Method::GET, &HashMap::new())
+        .await
+    {
+        Ok((status, body)) => (status, body).into_response(),
+        Err(_) => (
+            StatusCode::INTERNAL_SERVER_ERROR,
+            "Internal Server Error".to_string(),
+        )
+            .into_response(),
+    }
+}
+
+pub async fn procedure_pause_handler(State(state): State<AdminState>) -> impl IntoResponse {
+    match state
+        .procedure_handler
+        .handle("/procedure-manager/pause", Method::POST, &HashMap::new())
+        .await
+    {
+        Ok((status, body)) => (status, body).into_response(),
+        Err(_) => (
+            StatusCode::INTERNAL_SERVER_ERROR,
+            "Internal Server Error".to_string(),
+        )
+            .into_response(),
+    }
+}
+
+pub async fn procedure_resume_handler(State(state): State<AdminState>) -> impl IntoResponse {
+    match state
+        .procedure_handler
+        .handle("/procedure-manager/resume", Method::POST, &HashMap::new())
+        .await
+    {
+        Ok((status, body)) => (status, body).into_response(),
+        Err(_) => (
+            StatusCode::INTERNAL_SERVER_ERROR,
+            "Internal Server Error".to_string(),
+        )
+            .into_response(),
+    }
+}
```

Comment:
Can we just call the methods directly, like `get_procedure_manager_status` and `resume_procedure_manager`?
The handler on port 3002 is going away anyway, so it probably makes more sense to use the methods themselves.

Reply:
I have one question how to aviod cyclic package dependency when calling the methods like `get_procedure_manager_status` directly?  currently those methods are in `meta_srv` and  it depedents on `servers` package.